### PR TITLE
Upgrade to Rails 7

### PR DESCRIPTION
**Pull request by tsmartt**
Closes https://github.com/brave-intl/creators-private-issues/issues/184

What originally turned into a simple "update a few deps" PR has morphed into updating to Rails 7 as the foundation for fixing these issues.

### CODE REVIEW

**Comment by diracdeltas** [comment]
@tsmartt lgtm but can we get `yarn audit` added to one of the github actions jobs so it's easy to tell if there are dependency vulns? 

same with `bundler audit` or whatever the equivalent is these days for ruby

File: public/creators-landing/package.json [hunk 1]
```diff
@@ -48,6 +48,9 @@
     "ansi-regex": "5.0.1",
     "shell-quote": "1.7.3",
     "ansi-html": "https://registry.npmjs.org/ansi-html-community/-/ansi-html-community-0.0.8.tgz",
-    "immer": "9.0.6"
+    "immer": "9.0.6",
+    "loader-utils": "1.4.2",
+    "json5": "2.2.2",
+    "minimatch": "3.0.5"
```
Comment (diracdeltas) [inline public/creators-landing/package.json:54]: shouldn't there be an update to yarn.lock in this directory too?

**Review by diracdeltas — CHANGES_REQUESTED** [review]
seems like this needs a yarn lock update in creators-landing. also please add `yarn audit` and the ruby gems equivalent (bundler audit?) to a github action so it gets flagged earlier

**Review by diracdeltas — APPROVED** [review]
still some audit issues (and the audit command should be fixed to run the later commands even if the first audit fails), but approving since this is an improvement.

**Comment by thypon** [comment]
Why did we updated the SVGs here? Can we run the SVGO optimizer to remove spurious inkscape stuff?

File: config/environments/test.rb [hunk 2]
```diff
@@ -9,29 +8,41 @@
...
 
-  # Show full error reports.
-  config.consider_all_requests_local = true
-  config.action_controller.perform_caching = true # false?
-  config.public_file_server.headers = {
-    "Cache-Control" => "public, max-age=172800"
-  }
-  config.cache_store = :memory_store, {size: 64.megabytes}
+  # Show full error reports and disable caching.
+  config.consider_all_requests_local       = true
```
[the remark this answers is not in the document]
Comment (bcaller) [reply config/environments/test.rb:43]: ignore due to test environment

[the remark this answers is not in the document]
Comment (thypon) [reply config/environments/test.rb:43]: We should probably add that to the `.semgrepignore` file in this repo

**Review by jlbyrne — APPROVED** [review]

**Review by jlbyrne — APPROVED** [review]

**Pull request merged**